sphinx_rtd_theme/layout.html

```diff
@@ -133,23 +135,32 @@
         <div class="wy-side-nav-search" {% if theme_style_nav_header_background %} style="background: {{theme_style_nav_header_background}}" {% endif %}>
           {%- block sidebartitle %}
 
-          {%- if logo and theme_logo_only %}
+          {%- if ( logo or logo_url ) and theme_logo_only %}
+
+          {%- if sphinx_version_info < (4, 0) -%}
+            {# the logo helper function was removed in Sphinx 6 and deprecated since Sphinx 4 #}
+            {# the master_doc variable was renamed to root_doc in Sphinx 4 (master_doc still exists in later Sphinx versions) #}
+            {%- if logo %}
             <a href="{{ pathto(master_doc) }}">
+              <img src="{{ pathto('_static/' + logo, 1) }}" class="logo" alt="{{ _('Logo') }}"/>
+            </a>
+            {%- endif %}
+          {# Behavior since Sphinx 4 #}
           {%- else %}
-            <a href="{{ pathto(master_doc) }}" class="icon icon-home"> {{ project }}
+            {% if logo_url %}
+              <a href="{{ pathto(root_doc) }}">
+                <img src="{{ logo_url }}" class="logo" alt="{{ _('Logo') }}"/>
+              </a>
+            {%- endif %}
```

Comment:
It seems like the conditional `if logo` is missing here now, correct?

Reply:
The `logo_url` is a replacement helper function that works since Sphinx 4. So it's not necessary here.

Reply:
I'm aware of that change, but it is necessary to inspect that here as it's possible to use both the `logo` and `logo_url` variables in Sphinx 4 and 5.

Reply:
*) correction: not a helper function but just a context var: https://github.com/sphinx-doc/sphinx/blob/c1f2a52d5d97f874d84e4d5421c99a566242af6b/sphinx/builders/html/__init__.py#L1252-L1257

Reply:
Your comments make more sense looking into the code:
https://github.com/sphinx-doc/sphinx/blob/v5.3.0/sphinx/builders/html/__init__.py#L1252-L1257

So support for Sphinx 4,5 + `logo` variable is indeed here. It's probably worth a comment here, as this conditional logic is already difficult to parse.